### PR TITLE
[Docs] Add SDK package readme + Add web focused documentation to the SDK repo

### PR DESCRIPTION
**Pull request by iamalwaysuncomfortable**
## Motivation

Now that the Aleo typescript SDK has been fully developed, documentation of how to deploy web applications is necessary to help users create useful web applications. This PR creates detailed readmes in both repositories meant to exist on both NPM and the Aleo Developer hub

[Rendered Version](https://github.com/AleoHQ/sdk/blob/f03ff340658b94349e63a3ec8264eed41a63ef0f/sdk/README.md)

### CODE REVIEW

File: sdk/README.md [hunk 1]
```diff
@@ -2,36 +2,545 @@
...
+
+To build the project from source, go to this project's root and execute:
+
+`npm install && npm run build`
+
+## Getting Started
+
+### Zero Knowledge Web App Examples
+
+A set of fully functional examples and accompying documentation for creating zero knowledge web apps can be found in
```
Comment (onetrickwolf) [inline sdk/README.md]:
```suggestion
A set of fully functional examples and accompanying documentation for creating zero knowledge web apps can be found in 
```

File: sdk/README.md [hunk 2]
```diff
@@ -2,36 +2,545 @@
...
+The key used to represent an identity of an individual user. This key is used to authorize zero 
+knowledge program execution.
+#### View Key
+This key is derived from the private key and can be used to identify all records and transaction data that
+belongs to an individual user.
+#### Compute Key
+A key that can be used to trustlessly run applications and generate transactions on a user's behalf.
+#### Address
+A public address that can be used to trustlessly identify a user in order for that user to receive official
+aleo credits or unique data defined by other zero-knolwedge Aleo programs.
```
Comment (onetrickwolf) [inline sdk/README.md]:
```suggestion
Aleo credits or unique data defined by other zero-knowledge Aleo programs.
```

File: sdk/README.md [hunk 3]
```diff
@@ -2,36 +2,545 @@
...
+
+1. **KeyProvider:** When programs execute, by default, they will synthesize the proving and verifying keys needed to
+make a zero knowledge proof. However, these keys are large and expensive to generate. For this reason, applications may
+want to store these keys and re-use them for future execution. The `KeyProvider` interface provides the ability for
+users of the SDK to provide their own key storage and retrieval mechanism. The SDK provides a default implementation
+of the `KeyProvider` interface via the `AleoKeyProvider` class. 
+2. **RecordProvider:** When programs execute, they will often need to read and write records. The `RecordProvider` 
+interface allows users of the SDK to provide their own record storage and retrieval mechanism. The SDK provides a
+default implementation of the `RecordProvider` interface via the `NetworkRecordProvider` class which searches the Aleo
+network for records uniquely belong to a user.
```
Comment (onetrickwolf) [inline sdk/README.md]:
```suggestion
network for records uniquely belonging to a user.
```

**Review by onetrickwolf — APPROVED** [review]
Just small spelling issues, but wow amazing documentation overhaul!

File: sdk/README.md [hunk 4]
```diff
@@ -1,37 +1,851 @@
...
+    end
+    user1--record3 \n owner: user2address \n balance: 4000u64-->t1[transfer_private_to_public]
+    t1-.record4 \n owner: user2address \n amount: 1000u64.->user1
+    t1--amount 3000u64-->m1
+```
+
+#### 3. transfer_public
+
+* Subtracts an amount `credits` stored in the `account` mapping of the `credits.aleo program`, and 
+adds that amount to the `account` mapping of the receiver. This function is %100 public and does not consume or generate
```
Comment (collinc97) [inline sdk/README.md]:
```suggestion
adds that amount to the `account` mapping of the receiver. This function is 100% public and does not consume or generate
```

**Pull request merged**